review/communication-react.api.md

```diff
@@ -0,0 +1,2130 @@
+    on(event: 'isSpeakingChanged', listener: IsSpeakingChangedListener): void;
+    on(event: 'callEnded', listener: CallEndedListener): void;
+    on(event: 'diagnosticChanged', listener: DiagnosticChangedEventListner): void;
+    on(event: 'error', listener: (e: AdapterError) => void): void;
+}
+
+// @public
+export type CallAdapterUiState = {
+    isLocalPreviewMicrophoneEnabled: boolean;
+    page: CallCompositePage;
+};
+
+// @public
+export const CallAgentProvider: (props: CallAgentProviderProps) => JSX.Element;
+
+// @public
+export interface CallAgentProviderProps {
+    // (undocumented)
+    callAgent?: CallAgent;
+    // (undocumented)
+    children: React_2.ReactNode;
+}
+
+// @public
+export interface CallAgentState {
+    displayName?: string;
+}
+
+// @public
+export const CallClientProvider: (props: CallClientProviderProps) => JSX.Element;
+
+// @public
+export interface CallClientProviderProps {
+    // (undocumented)
+    callClient: StatefulCallClient;
+    // (undocumented)
+    children: React_2.ReactNode;
+}
+
+// @public
+export interface CallClientState {
+    callAgent?: CallAgentState;
+    calls: {
+        [key: string]: CallState;
+    };
+    callsEnded: {
+        [key: string]: CallState;
+    };
+    deviceManager: DeviceManagerState;
+    incomingCalls: {
+        [key: string]: IncomingCallState;
+    };
+    incomingCallsEnded: {
+        [key: string]: IncomingCallState;
+    };
+    latestErrors: CallErrors;
+    userId: CommunicationIdentifierKind;
+}
+
+// @public
+export const CallComposite: (props: CallCompositeProps) => JSX.Element;
+
+// @public
+export type CallCompositeIcons = Partial<Pick<CompositeIcons, 'ControlButtonCameraOff' | 'ControlButtonCameraOn' | 'ControlButtonEndCall' | 'ControlButtonMicOff' | 'ControlButtonMicOn' | 'ControlButtonOptions' | 'ControlButtonParticipants' | 'ControlButtonScreenShareStart' | 'ControlButtonScreenShareStop' | 'ErrorBarCallCameraAccessDenied' | 'ErrorBarCallCameraAlreadyInUse' | 'ErrorBarCallLocalVideoFreeze' | 'ErrorBarCallMacOsCameraAccessDenied' | 'ErrorBarCallMacOsMicrophoneAccessDenied' | 'ErrorBarCallMicrophoneAccessDenied' | 'ErrorBarCallMicrophoneMutedBySystem' | 'ErrorBarCallNetworkQualityLow' | 'ErrorBarCallNoMicrophoneFound' | 'ErrorBarCallNoSpeakerFound' | 'HorizontalGalleryLeftButton' | 'HorizontalGalleryRightButton' | 'LobbyScreenConnectingToCall' | 'LobbyScreenWaitingToBeAdmitted' | 'LocalDeviceSettingsCamera' | 'LocalDeviceSettingsMic' | 'LocalDeviceSettingsSpeaker' | 'LocalPreviewPlaceholder' | 'Muted' | 'NetworkReconnectIcon' | 'NoticePageAccessDeniedTeamsMeeting' | 'NoticePageJoinCallFailedDueToNoNetwork' | 'NoticePageLeftCall' | 'NoticePageRemovedFromCall' | 'OptionsCamera' | 'OptionsMic' | 'OptionsSpeaker' | 'ParticipantItemMicOff' | 'ParticipantItemOptions' | 'ParticipantItemOptionsHovered' | 'ParticipantItemScreenShareStart' | 'VideoTileMicOff'>>;
+
+// @public
+export type CallCompositeOptions = {
+    mobileView?: boolean;
```

Comment:
It's a general API design technique to avoid boolean flags to prevent future breaking changes.
For example, one can always refactor `x: boolean` to `foo: "x"`. That way this can evolve to `foo: "x" | "y"`. With boolean flags we would have to introduce a new flag and then we would have `x: boolean; y: boolean`. The addition of additional flags then raises the question which true/false combinatorics of them are valid configurations.

With all that preface why bools are hard to evolve:
Should we consider
```typescript
formFactor?: "desktop" | "mobile";
```

Reply:
The same bool to union type redesign should be considered everywhere where we use bool flags. (Not saying all need to be refactored, but all should get a second look).

Reply:
I like this a lot more, I'll update the mobileView flag to formFactor

Reply:
https://github.com/Azure/communication-ui-library/pull/1095